### PR TITLE
Opsgenie bug 46058

### PR DESCRIPTION
**Pull request by JudahSchwartz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/46058)

## Description
Adds polling when calling get_request 

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml [hunk 1]
```diff
@@ -1500,6 +1516,29 @@ script:
     - contextPath: OpsGenie.Team.name
       description: Team name.
       type: String
+  - name: opsgenie-get-request
+    polling: true
+    description: Get a request in Opsgenie
+    deprecated: false
```
Comment (bziser) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml:1522]: We want to expose `opsgenie-get-request` to the user?

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3_test.py [hunk 2]
```diff
@@ -919,3 +928,20 @@ def test_responders_to_json_empty_value():
     res = mock_client.responders_to_json(responders={},
                                          responder_key="responder")
     assert (res == {})
+
+
+def test_get_request_command(requests_mock, mocker):
```
Comment (bziser) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3_test.py:933]: Please add     
"""
    Given:
    When:
    Then:
    """

**Review by bziser — APPROVED** [review]

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py [hunk 3]
```diff
@@ -1015,7 +941,8 @@ def fetch_incidents_command(client: Client,
                                                                                      tags,
                                                                                      client.list_incidents,
                                                                                      now,
-                                                                                     demisto.get(last_run, f"{INCIDENT_TYPE}"))
+                                                                                     demisto.get(last_run,
```
Comment (Itay4) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py:944]: if not needed, please revert

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml [hunk 4]
```diff
@@ -1500,6 +1516,29 @@ script:
     - contextPath: OpsGenie.Team.name
       description: Team name.
       type: String
+  - name: opsgenie-get-request
+    polling: true
```
Comment (Itay4) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml:1520]: why should it be set to `polling` as well?

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py [hunk 5]
```diff
@@ -890,13 +794,35 @@ def remove_tag_incident(client: Client, args: Dict[str, Any]) -> CommandResults:
...
+            scheduled_command=ScheduledCommand(command='opsgenie-get-request',
+                                               next_run_in_seconds=int(
+                                                   args.get('interval_in_seconds', DEFAULT_POLL_INTERVAL)),
+                                               args={**args, 'polled_once': True},
+                                               timeout_in_seconds=int(
+                                                   args.get('timeout_in_seconds', DEFAULT_POLL_TIMEOUT))))
+    else:
+        results_dict = results.json()
+        return CommandResults(
+            outputs_prefix=args.get("output_prefix", f'OpsGenie.{request_type.capitalize()[:-1]}'),
```
Comment (Itay4) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py:821]: arent we changing the context path here?

File: Packs/OpsGenie/ReleaseNotes/2_0_3.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### OpsGenie v3
+- Fixed an issue where commands failed with a 404 error.
```
Comment (Itay4) [inline Packs/OpsGenie/ReleaseNotes/2_0_3.md:4]: need to improve the release notes here and mention that polling is now used

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml [hunk 7]
```diff
@@ -1500,6 +1516,31 @@ script:
...
+        isArray: false
+      - name: request_type
+        description: The type of the request to get
+        required: true
+        isArray: false
+        auto: PREDEFINED
+        predefined:
+          - alerts
+          - incedents
+    outputs: []
```
Comment (Itay4) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml:1543]: we do have have outputs, no?

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py [hunk 8]
```diff
@@ -27,21 +31,15 @@ class Client(BaseClient):
...
-        if not demisto.get(res, "data.success"):
-            status = demisto.get(res, "data.status")
-            raise DemistoException(f"The command failed, the reason: {status}")
-        return res
+        return self._http_request(
+            method='GET',
+            url_suffix=f"{url_suffix}/{args.get('request_type')}/{REQUESTS_SUFFIX}/"
+                       f"{args.get('request_id')}",
+            ok_codes=(404, 200),
+            resp_type='all')
```
Comment (Itay4) [inline Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py:42]:
```suggestion
            resp_type='response')
```

File: Packs/OpsGenie/Integrations/OpsGenieV3/README.md [hunk 9]
```diff
@@ -1660,6 +1660,65 @@ Get teams
...
+| --- | --- | --- |
+| request_id | The id of the request to get | Required | 
+| request_type | The type of the request to get | Required |  
+
+
+#### Context Output
+
+| **Path**                              | **Type** | **Description** |
+|---------------------------------------| --- | --- |
+| OpsGenie.[request_type].action        | String | Action of this request. |
```
Comment (Itay4) [inline Packs/OpsGenie/Integrations/OpsGenieV3/README.md:1682]: need to replace the `request_type` with both options

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml (hunk 4 above)
Comment (JudahSchwartz) [reply Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml:1520]: any command which can return a scheduled result needs polling: true

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml (hunk 7 above)
Comment (JudahSchwartz) [reply Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml:1543]: Outputs depend on the input. Should both be listed here?

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py (hunk 5 above)
Comment (JudahSchwartz) [reply Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py:821]: yes, depending on what the type is

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py (hunk 3 above)
Comment (JudahSchwartz) [reply Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py:944]: was getting an error from linter. Can this be merged anyway?

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py (hunk 5 above)
Comment (Itay4) [reply Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py:821]: so we're breaking bc?

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml (hunk 7 above)
Comment (Itay4) [reply Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml:1543]: lets discuss

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py (hunk 5 above)
Comment (JudahSchwartz) [reply Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.py:821]: no this is in order to preserve the bc. We had copy pasted code before that would return according to the type

File: Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml (hunk 7 above)
Comment (Itay4) [reply Packs/OpsGenie/Integrations/OpsGenieV3/OpsGenieV3.yml:1543]: yes

**Pull request merged**